src/core_modules/capture-core/components/Pages/NewRelationship/RegisterTei/GeneralOutput/WarningsSection/SearchGroupDuplicate/searchGroupDuplicate.epics.js

```diff
@@ -71,17 +71,32 @@ export const loadSearchGroupDuplicatesForReviewEpic = (action$: InputObservable,
                 ouMode: 'ACCESSIBLE',
                 pageSize: 5,
                 page: requestPage,
-                totalPages: !isChangePage,
                 filter: filters,
                 ...contextParam,
             };
             const attributes = contextParam.program ?
                 getTrackerProgramThrowIfNotFound(contextParam.program).attributes :
                 getTrackedEntityTypeThrowIfNotFound((contextParam.trackedEntityType)).attributes;
 
-            return from(getTrackedEntityInstances(queryArgs, attributes)).pipe(
-                map(({ trackedEntityInstanceContainers: searchResults, pagingData }) =>
-                    duplicatesForReviewRetrievalSuccess(searchResults, pagingData)),
+            return forkJoin(
+                {
+                    first: getTrackedEntityInstances(queryArgs, attributes),
+                    second: getTrackedEntityInstances({ ...queryArgs, filter: filters, page: requestPage + 1 }, attributes),
+                },
+            ).pipe(
+                map(({
+                    first: { trackedEntityInstanceContainers: searchResults, pagingData },
+                    second: { trackedEntityInstanceContainers: nextPageSearchResults },
+                }) => {
+                    const isPaginationButtonDisabled = Boolean(nextPageSearchResults.length === 0);
+                    return duplicatesForReviewRetrievalSuccess(
+                        searchResults,
+                        {
+                            currentPage: pagingData.currentPage,
+                            isPaginationButtonDisabled,
+                        },
+                    );
+                }),
```

Comment:
Using `forkJoin` we can perform multiple requests at a time. We get the Observable back when they are completed. 

I think this is handling gracefully the limitation we face with the backend at the moment. We dont ask the user to click the next paging button to show them an empty page but we do it for them under the hood.

I applied the changes with the forkJoin in all the places that search takes place.